### PR TITLE
TASK: improve dimension space point documentation

### PR DESCRIPTION
**Pull request by mhsdesign**
**Upgrade instructions**

**Review instructions**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

**Review by dlubitz — APPROVED** [review]
I don't see the need for the `zeroDimensional` named constructors. You will never use them as usually the dimension configuration is dynamic and loaded from configuration.

But I'm fine with having them there, if it makes the usage more clear.

File: Neos.ContentRepository.Core/Classes/DimensionSpace/DimensionSpacePoint.php [hunk 1]
```diff
@@ -61,10 +61,19 @@ public static function fromJsonString(string $jsonString): self
         return self::instance(json_decode($jsonString, true));
     }
 
+    /**
+     * Creates a dimension space point for a simple zero-dimensional content repository.
+     */
+    public static function zeroDimensional(): self
```
Comment (bwaidelich) [inline Neos.ContentRepository.Core/Classes/DimensionSpace/DimensionSpacePoint.php]: Thank you for this, this is so much more explicit than the `fromArray([])` calls.
I just wonder about the naming.
In other places we use `create...` (for example `createEmpty()`) to emphasize that this is a factory method..

OTOH and "empty" DSP is maybe not much clearer..

Comment (mficzel) [reply Neos.ContentRepository.Core/Classes/DimensionSpace/DimensionSpacePoint.php]: Suggestion: `createWithoutDimensions`

File: Neos.ContentRepository.Core/Classes/DimensionSpace/DimensionSpacePoint.php [hunk 2]
```diff
@@ -61,10 +61,19 @@ public static function fromJsonString(string $jsonString): self
         return self::instance(json_decode($jsonString, true));
     }
 
+    /**
+     * Creates a dimension space point for a simple zero-dimensional content repository.
```
Comment (ahaeslich) [inline Neos.ContentRepository.Core/Classes/DimensionSpace/DimensionSpacePoint.php]: @nezaniel and @mhsdesign will update this doc block

File: Neos.ContentRepository.Core/Classes/DimensionSpace/DimensionSpacePointSet.php [hunk 3]
```diff
@@ -14,9 +14,14 @@
 
 namespace Neos\ContentRepository\Core\DimensionSpace;
 
+use Neos\ContentRepository\Core\EventStore\EventInterface;
```
Comment (mficzel) [inline Neos.ContentRepository.Core/Classes/DimensionSpace/DimensionSpacePointSet.php:17]: Why is this here?

**Review by mficzel — APPROVED** [review]

**Comment by mficzel** [comment]
fine by 👀

File: Neos.ContentRepository.Core/Classes/DimensionSpace/DimensionSpacePointSet.php (hunk 3 above)
Comment (mhsdesign) [reply Neos.ContentRepository.Core/Classes/DimensionSpace/DimensionSpacePointSet.php:17]: for the docblock ;)
